### PR TITLE
Adding customer storage recalculator and entity counter recalculator functions

### PR DESCRIPTION
**Pull request by JackLewis-digirati**
This PR adds in  docker based python images that can be run with AWS lambda

### CODE REVIEW

File: .github/actions/docker-build-and-push/action.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    required: true
+  github-token:
+    description: "The GitHub token used when interacting with GCHR."
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - id: checkout
+      uses: actions/checkout@v2
```
Comment (donaldgray) [inline .github/actions/docker-build-and-push/action.yml]: `actions/checkout@v3`

File: .github/actions/docker-build-and-push/action.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    - id: docker-setup-buildx
+      uses: docker/setup-buildx-action@v2
+      with:
+        driver-opts: |
+          image=moby/buildkit:v0.10.6
```
Comment (donaldgray) [inline .github/actions/docker-build-and-push/action.yml]: Is this still required? It was related to a ghcr/buildkit issue from a while ago and I'm wondering if it's fixed yet (https://github.com/docker/build-push-action/issues/761#issuecomment-1398918693)

File: sql/deleteSql.sql [hunk 3]
```diff
@@ -0,0 +1,60 @@
+-- select customer images for deletion
```
Comment (donaldgray) [inline sql/deleteSql.sql:1]: Is this file a duplicate of `deleteSql.sql`?

File: src/customer_storage_recalculator.py [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+        logger.debug(f"updating cloudwatch metrics - {metric_data}")
+        cloudwatch.put_metric_data(MetricData=metric_data,
+                                   Namespace='Entity Counter Recalculator')
+        return metric_data
+    except Exception as e:
+        logger.error(f"Error posting to cloudwatch: {e}")
+        raise e
+
+
+def __run_sql(conn):
```
Comment (donaldgray) [inline src/customer_storage_recalculator.py:91]: Could you calculate space level changes first then customer level? Would save multiple scans of `"ImageStorage"`

File: src/entity_counter_recalculator.py [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+        logger.debug(f"updating cloudwatch metrics - {metric_data}")
+        cloudwatch.put_metric_data(MetricData=metric_data,
+                                   Namespace='Entity Counter Recalculator')
+        return metric_data
+    except Exception as e:
+        logger.error(f"Error posting to cloudwatch: {e}")
+        raise e
+
+
+def __run_sql(conn):
```
Comment (donaldgray) [inline src/entity_counter_recalculator.py:87]: Similar to customer-storage, could `space-images` be worked out then `customer-images` by suming those?

**Review by donaldgray — APPROVED** [review]
Couple of comments but other than that LGTM.

_Edit_
Is it worth squashing a few of the commits for a cleaner commit history? E.g.
* fixing yaml
* fixing pip install
* attempting dependencies zip
* changing to use ${{NOW}}
* changing to use $(NOW)
* changing to use ${{ env.NOW }}
* removing colons
* changing zip package
* removing /packages 

into - "fixing various issues with zip" (bad naming but hopefully you get what I mean!)

File: sql/deleteSql.sql (hunk 3 above)
Comment (JackLewis-digirati) [reply sql/deleteSql.sql:1]: apologies, must have slipped through! I've removed one

**Pull request merged**